### PR TITLE
Allow motherduck and in-memory duckdb connections

### PR DESCRIPTION
**Pull request by archiewood**
### Checklist

- [ ] For UI or styling changes, I have added a screenshot or gif showing before & after
- [x] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)

### CODE REVIEW

File: packages/duckdb/index.cjs [hunk 1]
```diff
@@ -185,17 +185,27 @@ module.exports.getRunner = async (opts, directory) => {
...
+		return runQuery(queryContent, { ...opts, filename: filename }, batchSize);
 	};
 };
 
 /** @type {import("@evidence-dev/db-commons").ConnectionTester<DuckDBOptions>} */
 module.exports.testConnection = async (opts, directory) => {
-	const r = await runQuery('SELECT 1;', { ...opts, filename: path.join(directory, opts.filename) })
+	// Determine filename based on string
+	const filename =
+		opts.filename.startsWith('md:?motherduck_token=') || opts.filename == ':memory:'
```
Comment (archiewood) [inline packages/duckdb/index.cjs:201]: a bit of copypasta here, should probably define elsewhere?

File: packages/duckdb/index.cjs [hunk 2]
```diff
@@ -185,17 +185,27 @@ module.exports.getRunner = async (opts, directory) => {
...
 		// Filter out non-sql files
 		if (!queryPath.endsWith('.sql')) return null;
-		return runQuery(
-			queryContent,
-			{ ...opts, filename: path.join(directory, opts.filename) },
-			batchSize
-		);
+		// Determine filename based on string
+		const filename =
+			opts.filename.startsWith('md:?motherduck_token=') || opts.filename == ':memory:'
```
Comment (archiewood) [inline packages/duckdb/index.cjs:190]: I wonder if a more robust way is to assume that it is an absolute path by default, and only revert to relative if certain conditions are met.

https://duckdb.org/docs/connect

It seems there are three possibilities in general
- An existing file
- In memory - `:memory:`
- If no file is specified, one will be created (the duckdb CLI creates an in-memory file , for example)

File: .changeset/grumpy-dancers-design.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@evidence-dev/duckdb': major
```
Comment (ItsMeBrianD) [inline .changeset/grumpy-dancers-design.md:2]: patch fix

File: packages/duckdb/index.cjs (hunk 1 above)
Comment (ItsMeBrianD) [reply packages/duckdb/index.cjs:201]: Is there any reason this can't live in `runQuery`?

**Review by ItsMeBrianD — CHANGES_REQUESTED** [review]
Not a major change, filename can probably be declared in `runQuery`

**Pull request merged**